### PR TITLE
Development

### PR DESCRIPTION
**Pull request by heitorhackcanabal**

### CODE REVIEW

File: app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerCollapsingToolbarLandscapeLayout.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public BannerCollapsingToolbarLandscapeLayout(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+
+        setMeasuredDimension(getMeasuredWidth(), Double.valueOf(getMeasuredWidth() * 0.3).intValue());
```
Comment (arctouch-kaiqueferreira) [inline app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerCollapsingToolbarLandscapeLayout.java]: What is the meaning of those constants you are multiplying here?

File: app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerLandscapeImageView.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package br.com.h2capps.upcomingmovies.ui.components;
+
+import android.content.Context;
+import android.util.AttributeSet;
+
+import com.loopj.android.image.SmartImageView;
+
```
Comment (arctouch-kaiqueferreira) [inline app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerLandscapeImageView.java]: The portrait and landscape versions of these classe have many in common. Is there a way to remove duplicated code here?

File: app/src/main/res/layout/activity_movie_details.xml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:background="@color/colorPrimary"
+                    android:scaleType="fitXY" />
+
+                <LinearLayout
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="3"
+                    android:layout_marginLeft="16dp"
```
Comment (arctouch-kaiqueferreira) [inline app/src/main/res/layout/activity_movie_details.xml]: I am happy you used material dimensions here :)

File: app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerCollapsingToolbarLandscapeLayout.java (hunk 1 above)
Comment (heitorhackcanabal) [reply app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerCollapsingToolbarLandscapeLayout.java]: I created this custom component to have the exact size of the banner image. It takes the size of the width to know the size of the height.

File: app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerLandscapeImageView.java (hunk 2 above)
Comment (heitorhackcanabal) [reply app/src/main/java/br/com/h2capps/upcomingmovies/ui/components/BannerLandscapeImageView.java]: The only difference is the proportion of the image, which in the landscape is smaller so it does not take up too much screen space. One way to remove duplicate code would be to pass by parameter to the class the multiplier value.

**Pull request merged**